### DISCUSSION TITLE
Pre-built generated types and source control

### DISCUSSION THREAD

**Discussion by leo-amelyan** [JasperFx/marten#2332] [Q&A] [answered]
The [documentation recommends](https://martendb.io/configuration/prebuilding.html#generating-all-types-upfront) that the generated code is not checked into source control.

I agree with the recommendation but don't see a way to include generated code to be available for Marten at Production time. 

I have tried to run the  `dotnet run -- codegen write` as a step in the Azure build Pipeline, however, that doesn't work because it tried to connect to Marten DB and fails. Also, it doesn't sound right to run the application in CI Pipeline. 

So, instead, I opted to check in the generated code so that it can be built into the assembly and available in Production.  

Is there another option to include generated files without putting them in source control?

**Comment by jeremydmiller** [comment, marked as answer]
> The [documentation recommends](https://martendb.io/configuration/prebuilding.html#generating-all-types-upfront) that the generated code is not checked into source control.

I think I wrote that long before we had the V5 optimized workflow, but I don't think I'd stress out too much about checking in the pre-generated code. Or even agree with that today. The only issue is that you may sometimes want to manually delete that code when you change Marten configuration.

But also: https://github.com/JasperFx/marten/issues/2335
